crates/wasmtime/src/engine.rs

```diff
@@ -239,6 +239,22 @@ impl Engine {
         Ok(mmap.to_vec())
     }
 
+    /// Returns a value that can be used to check precompiled WebAssembly compatibility.
+    ///
+    /// The outputs of [`Engine::precompile_module`] and [`Engine::precompile_component`]
+    /// are compatible with a different [`Engine`] instance only if the two engines use
+    /// compatible [`Config`]s. The value returned by this function will be the same for two
+    /// [`Engine`]s if a binary from one can be deserialized by the other.
+    #[cfg(compiler)]
+    #[cfg_attr(nightlydoc, doc(cfg(feature = "cranelift")))] // see build.rs
+    pub fn precompile_compatibility_key(&self) -> impl AsRef<[u8]> {
+        use std::hash::{BuildHasher, Hasher};
+        let data = serialization::build_compiler_info(&self);
+        let hasher = hashbrown::hash_map::DefaultHashBuilder::default().build_hasher();
```

Comment:
I think a non-crypto hash is safe enough here. If an attacker can control enough of the build_compiler_info to force a collision you probably are already done for.

Reply:
Hashbrown uses ahash which doesn't guarantee stability between different systems. In fact it returns different values between systems with hardware AES instructions and those that don't. As such using ahash here will break cross-compilation.

Reply:
Ah good to know, thanks.

Reply:
Replaced with sha2, which is probably where I should have started anyway. Its a new dep under some feature sets but it is already used by the cache system.